HDLC address/control octet skip patch - ID: 1596730
Don Stokes ( don_stokes ) - 2006-11-15 01:54
http://sourceforge.net/tracker/?func=detail&aid=1596730&group_id=63352&atid=503661

diff -BurpN gateway/rp-l2tp-0.4/dgram.c gateway/rp-l2tp/dgram.c
--- gateway/rp-l2tp-0.4/dgram.c	2009-09-02 16:37:57.000000000 +0000
+++ gateway/rp-l2tp/dgram.c	2009-09-02 16:38:51.000000000 +0000
@@ -378,8 +378,10 @@ l2tp_dgram_take_from_wire(struct sockadd
 	}
 
 	/* Forget the 0xFF, 0x03 HDLC markers */
-	payload += 2;
-	framelen -= 2;
+	if(payload[0] == 0xff && payload[1] == 0x03) {
+	    payload += 2;
+	    framelen -= 2;
+	}
 
 	if (framelen < 0) {
 	    continue;
